gobblin-data-management/src/main/java/org/apache/gobblin/data/management/copy/publisher/CopyDataPublisher.java

```diff
@@ -185,27 +191,35 @@ private static Multimap<CopyEntity.DatasetAndPartition, WorkUnitState> groupByFi
    * and versionStrategy (usually relevant to modTime as well), since they are subject to change with Publish(rename)
    */
   private void preserveFileAttrInPublisher(CopyableFile copyableFile) throws IOException {
-    // Preserving File ModTime, and set the access time to an initializing value when ModTime is declared to be preserved.
-    if (copyableFile.getPreserve().preserve(PreserveAttributes.Option.MOD_TIME)) {
-      fs.setTimes(copyableFile.getDestination(), copyableFile.getOriginTimestamp(), -1);
+    if (!copyableFile.getFileStatus().isFile() && state.getProp(DatasetUtils.DATASET_PROFILE_CLASS_KEY, "").equals(
+        ManifestBasedDatasetFinder.class.getCanonicalName())){
```

Comment:
I see the reason why, but it's really an anti-pattern to determine behavior based on hard-coded class names.  seems even less desirable to do so in an unrelated class (as this class doesn't otherwise know about datasets or dataset finders).

might there be another way, such as inserting a config flag, with a clear k-v pair (other than the name of this distant impl. class)?  e.g. `alwaysOverwriteDirMetadata`?